openedx/core/djangoapps/schedules/tasks.py

```diff
@@ -61,58 +47,133 @@ def update_course_schedules(self, **kwargs):
         raise self.retry(kwargs=kwargs, exc=exc)
 
 
-class RecurringNudge(ScheduleMessageType):
-    def __init__(self, day, *args, **kwargs):
-        super(RecurringNudge, self).__init__(*args, **kwargs)
-        self.name = "recurringnudge_day{}".format(day)
+class ScheduleMessageBaseTask(Task):
+    ignore_result = True
+    routing_key = ROUTING_KEY
+    num_bins = resolvers.DEFAULT_NUM_BINS
+    enqueue_config_var = None  # define in subclass
+    log_prefix = None
+    resolver = None  # define in subclass
+    async_send_task = None  # define in subclass
+
+    @classmethod
+    def log_debug(cls, message, *args, **kwargs):
+        LOG.debug(cls.log_prefix + ': ' + message, *args, **kwargs)
```

Comment:
Could just use the `PrefixedDebugLoggerMixin`?

Reply:
This version is intended as a `classmethod`, and that one is an instance method.